src/providers/postgres/qgspostgresfeatureiterator.cpp

```diff
@@ -474,6 +474,13 @@ bool QgsPostgresFeatureIterator::close()
 QString QgsPostgresFeatureIterator::whereClauseRect()
 {
   QgsRectangle rect = mFilterRect;
+
+  // in case coordinates are around 180.0°
+  if ( rect.xMinimum() > rect.xMaximum() )
```

Comment:
I think this should also be testing that the geometries ARE geographic, or it'll potentially cause issues later (if we add support for non-geographic wrap arounds of the date line)

Reply:
Sure! But what is the best? `mSource->mCrs.isGeographic()` or `mSource->mSpatialColType == SctGeography`